### DISCUSSION TITLE
Bazel Installation Error

### DISCUSSION THREAD

**Discussion by theMr17** [oppia/oppia-android#5241] [Q&A (Installation)] [answered]
### Operating System

Windows 11

### Android Studio Version

Android Studio Giraffe 2022.3.1 Patch 3

### RAM size

16 GB

### Free disk space

100 GB

### Which step are you stuck on?

I am stuck on [Step 4: Installing Bazel](https://github.com/oppia/oppia-android/wiki/Bazel-Setup-Instructions-for-Windows#4-installing-bazel). I have done all the prior steps successfully.

### Error Log

```bash
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package bazel-4.0.0
E: Couldn't find any package by glob 'bazel-4.0.0'
```

**Comment by Rd4dev** [comment, marked as answer]
Hi @theMr17,

**Updated Answer:**

While following the [Bazel Installation Steps](https://github.com/oppia/oppia-android/wiki/Bazel-Setup-Instructions-for-Windows#4-installing-bazel), I missed out performing the [one-time setup](https://bazel.build/install/ubuntu#add-dis-uri) that is required before performing the installation command. 

Make sure to complete the one-time setup by following:

```sh
sudo apt install apt-transport-https curl gnupg -y
curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor >bazel-archive-keyring.gpg
sudo mv bazel-archive-keyring.gpg /usr/share/keyrings
echo "deb [arch=amd64 signed-by=/usr/share/keyrings/bazel-archive-keyring.gpg] https://storage.googleapis.com/bazel-apt stable jdk1.8" | sudo tee /etc/apt/sources.list.d/bazel.list
```

and then install bazel using:

```sh
sudo apt install bazel-4.0.0
```

and set bazel to the specified version by creating a symlink:

```sh
sudo ln -s /usr/bin/bazel-4.0.0 /usr/bin/bazel
```

Following this I was able to install bazel version 4.0.0

**Original Answer:**

I tried to follow the provided steps for installing Bazel but was encountered with the same issue.
![image](https://github.com/oppia/oppia-android/assets/122200035/2964c718-e337-48c7-9755-80897f9f0667)

So after a little bit of trial and error I was able to install it.

**Step 1:**
Got the link for the binary installer for Linux from [Bazel Github Releases](https://github.com/bazelbuild/bazel/releases/tag/4.0.0)
In your WSL terminal, use the wget command to download the installer script. 
Note: Move to a directory where your user has write permissions. For example, you can change to your home directory:
**home directory:**
`$ cd ~`
**wget command:**
`$ wget https://github.com/bazelbuild/bazel/releases/download/4.0.0/bazel-4.0.0-installer-linux-x86_64.sh`

![image](https://github.com/oppia/oppia-android/assets/122200035/930a045a-d3bb-4050-99e7-85b2ef8f9a8d)

**Step 2:**
If the download is successful, proceed to make the script executable.
`$ chmod +x bazel-4.0.0-installer-linux-x86_64.sh`

**Step 3:**
Run the installer script.
`$ ./bazel-4.0.0-installer-linux-x86_64.sh --user`

![image](https://github.com/oppia/oppia-android/assets/122200035/70dc1098-4b80-4dd5-98af-76d7847ece46)

**Step 4:**
Follow the instructions provided by the installer to add Bazel to your _PATH_.

**Verify Installation:**
Open a new WSL terminal and run:
`$ bazel version`

![image](https://github.com/oppia/oppia-android/assets/122200035/fdb69d54-17b4-4cae-9e84-b5f0f41430be)

The outlined steps successfully facilitated the installation of Bazel on my device. If you encounter any challenges or issues during the process, please don't hesitate to reach out. I'll be ready to assist and address any concerns promptly.

Regards,
**RD** (Rama Devi)

**Reply by theMr17** [reply]
Thanks @Rd4dev, that works perfectly!
